src/gerbil/core/contract.ss

```diff
@@ -635,6 +651,71 @@ package: gerbil/core
        #'(%%app arg ...))))
 
   (defsyntax (%%ref-dotted stx)
+    (def (make-method-lambda Interface-method object sig)
+      (def (lambda-list)
+	;; This turns a `(name args ... := default)` or `(name
+	;; default)` or `(name args ...)` contract interface argument
+	;; (AKA carg) specification into a lambda-list compatible `(name
+	;; default)` or just `name` argument.
+	(def (carg->lambda-arg carg)
+	  (with ((cons name rest) carg)
+	    (let (val (if (= (length rest) 1) (car rest)
+			  (let (:= (memq ':= rest))
+			    (if := (cadr :=) absent-obj))))
+	      (if (eq? val absent-obj) name [name val]))))
```

Comment:
it is better to use `absent-value` or just `#!void`, as `absent-obj` has some deleterious semantics deep in gambit.